src/main/java/io/openliberty/tools/intellij/lsp4mp/lsp4ij/server/ProcessStreamConnectionProvider.java

```diff
@@ -160,5 +170,42 @@ public String toString() {
         return "ProcessStreamConnectionProvider [commands=" + this.getCommands() + ", workingDir=" //$NON-NLS-1$//$NON-NLS-2$
                 + this.getWorkingDirectory() + "]"; //$NON-NLS-1$
     }
+    public void checkJavaHome(String javaHome, String serverType) {
+
+        if (javaHome == null) {
+            String errorMessage = LocalizedResourceUtil.getMessage("liberty.action.selection.dialog.java-home.null",serverType);
+            LOGGER.error(errorMessage);
+            showErrorPopup(errorMessage);
+            return;
+        }
+
+        File javaHomeDir = new File(javaHome);
+        if (!javaHomeDir.exists()) {
+            String errorMessage = LocalizedResourceUtil.getMessage("liberty.action.selection.dialog.javaHomeDir.exist",serverType,Constants.REQUIRED_JAVA_VERSION,javaHome);
+            LOGGER.error(errorMessage);
+            showErrorPopup(errorMessage);
+            return;
+        }
+
+        if (!checkJavaVersion(javaHome, Constants.REQUIRED_JAVA_VERSION)) {
+            String errorMessage = LocalizedResourceUtil.getMessage("liberty.action.selection.dialog.java-version", serverType,Constants.REQUIRED_JAVA_VERSION,javaHome);
```

Comment:
The value of `java.home` is being passed a parameter but is not being used in the error message. Please verify that the message contains all the expected parameters.